protocol/handshaking/serverbound_handshake_test.go

```diff
@@ -1,6 +1,346 @@
 package handshaking
 
-import "testing"
+import (
+	"bytes"
+	"github.com/haveachin/infrared/protocol"
+	"net"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestServerBoundHandshake_Marshal(t *testing.T) {
+	tt := []struct {
+		packet          ServerBoundHandshake
+		marshaledPacket protocol.Packet
+	}{
+		{
+			packet: ServerBoundHandshake{
+				ProtocolVersion: 578,
+				ServerAddress:   "spook.space",
+				ServerPort:      25565,
+				NextState:       ServerBoundHandshakeStatusState,
+			},
+			marshaledPacket: protocol.Packet{
+				ID:   0x00,
+				Data: []byte{0xC2, 0x04, 0x0B, 0x73, 0x70, 0x6F, 0x6F, 0x6B, 0x2E, 0x73, 0x70, 0x61, 0x63, 0x65, 0x63, 0xDD, 0x01},
+			},
+		},
+		{
+			packet: ServerBoundHandshake{
+				ProtocolVersion: 578,
+				ServerAddress:   "example.com",
+				ServerPort:      1337,
+				NextState:       ServerBoundHandshakeStatusState,
+			},
+			marshaledPacket: protocol.Packet{
+				ID:   0x00,
+				Data: []byte{0xC2, 0x04, 0x0B, 0x65, 0x78, 0x61, 0x6D, 0x70, 0x6C, 0x65, 0x2E, 0x63, 0x6F, 0x6D, 0x05, 0x39, 0x01},
+			},
+		},
+	}
+
+	for _, tc := range tt {
+		pk := tc.packet.Marshal()
+
+		if pk.ID != ServerBoundHandshakePacketID {
+			t.Error("invalid packet id")
+		}
+
+		if !bytes.Equal(pk.Data, tc.marshaledPacket.Data) {
+			t.Errorf("got: %v, want: %v", pk.Data, tc.marshaledPacket.Data)
+		}
+	}
+}
+
+func TestUnmarshalServerBoundHandshake(t *testing.T) {
+	tt := []struct {
+		packet             protocol.Packet
+		unmarshalledPacket ServerBoundHandshake
+	}{
+		{
+			packet: protocol.Packet{
+				ID:   0x00,
+				Data: []byte{0xC2, 0x04, 0x0B, 0x73, 0x70, 0x6F, 0x6F, 0x6B, 0x2E, 0x73, 0x70, 0x61, 0x63, 0x65, 0x63, 0xDD, 0x01},
+			},
+			unmarshalledPacket: ServerBoundHandshake{
+				ProtocolVersion: 578,
+				ServerAddress:   "spook.space",
+				ServerPort:      25565,
+				NextState:       ServerBoundHandshakeStatusState,
+			},
+		},
+		{
+			packet: protocol.Packet{
+				ID:   0x00,
+				Data: []byte{0xC2, 0x04, 0x0B, 0x65, 0x78, 0x61, 0x6D, 0x70, 0x6C, 0x65, 0x2E, 0x63, 0x6F, 0x6D, 0x05, 0x39, 0x01},
+			},
+			unmarshalledPacket: ServerBoundHandshake{
+				ProtocolVersion: 578,
+				ServerAddress:   "example.com",
+				ServerPort:      1337,
+				NextState:       ServerBoundHandshakeStatusState,
+			},
+		},
+	}
+
+	for _, tc := range tt {
+		hs, err := UnmarshalServerBoundHandshake(tc.packet)
+		if err != nil {
+			t.Error(err)
+		}
+
+		if hs.ProtocolVersion != tc.unmarshalledPacket.ProtocolVersion ||
+			hs.ServerAddress != tc.unmarshalledPacket.ServerAddress ||
+			hs.ServerPort != tc.unmarshalledPacket.ServerPort ||
+			hs.NextState != tc.unmarshalledPacket.NextState {
+			t.Errorf("got: %v, want: %v", hs, tc.unmarshalledPacket)
+		}
```

Comment:
```suggestion
                up := tc.unmarshalledPacket
                
		if hs.ProtocolVersion != up.ProtocolVersion ||
			hs.ServerAddress != up.ServerAddress ||
			hs.ServerPort != up.ServerPort ||
			hs.NextState != up.NextState {
			t.Errorf("got: %v, want: %v", hs, up)
		}
```
Just makes it more readable